packages/calypso-e2e/src/lib/pages/published/published-post-page.ts

```diff
@@ -39,51 +30,52 @@ export class PublishedPostPage {
 		await Promise.all( [ this.page.waitForNavigation(), this.page.keyboard.press( 'Enter' ) ] );
 	}
 
-	/**
-	 * Returns the frame which holds the Like widget.
-	 *
-	 * @returns {Promise<Frame>} Frame holding the like widget on page.
-	 */
-	private async getLikeFrame(): Promise< Frame > {
-		// Obtain the ElementHandle for the widget containing the like/unlike button.
-		const elementHandle = await this.page.waitForSelector( selectors.likeWidget );
-		// Without the next line, headless viewports fail to locate the Like widget.
-		await elementHandle.scrollIntoViewIfNeeded();
-		// Obtain the Frame object from the elementHandleHandle. This represents the widget iframe.
-		const frame = ( await elementHandle.contentFrame() ) as Frame;
-		// Wait until the widget element is stable in the DOM.
-		await elementHandle.waitForElementState( 'stable' );
-
-		return frame;
-	}
-
 	/**
 	 * Clicks the Like button on the post.
 	 *
 	 * This method will also confirm that click action on the Like button
 	 * had the intended effect.
-	 *
-	 * @returns {Promise<void>} No return value.
 	 */
 	async likePost(): Promise< void > {
-		await this.page.evaluate( () => window.scrollTo( 0, document.body.scrollHeight ) );
-		const frame = await this.getLikeFrame();
-		await frame.click( selectors.likeButton );
-		await frame.waitForSelector( selectors.likedText, { state: 'visible' } );
+		const locator = this.page
+			.frameLocator( 'iframe[title="Like or Reblog"]' )
+			.getByRole( 'link', { name: 'Like' } );
```

Comment:
Let's use the `exact` match here since `Like` can match `Liked`. This is important, especially because we're retrying `likePost` which can accidentally unlike on a subsequent call without the exact match.

```suggestion
			.getByRole( 'link', { name: 'Like', exact: true } );
```

Reply:
Definitely. Fixed in eea8d6bcc30f4acb9b1e6def75ef20a599472aca.